dataflow_test.py

```diff
@@ -452,3 +452,33 @@ def test_slice_raises_ValueError(args):
 @mark.xfail
 def test_pipes_must_end_in_a_sink():
     raise NotImplementedError
+
+
+def test_count_filter_filters_correctly():
+    even_numbers = list(range(0, 20, 2))
+    odd_numbers  = list(range(1, 20, 2))
+    the_source   = odd_numbers + even_numbers
```

Comment:
It makes me a little nervous that they are not interleaved at all.

Reply:
I thought about that... given the simplicity of the filtering maybe it is enough to use the continuous sequence and split them for the comparison afterwards